src/Dashboard/Dashboard.js

```diff
@@ -10,166 +15,75 @@ import TodoFilter from "../TodoFilter/TodoFilter";
-
-      setTodoArray(updatedGroupArray);
-   };
-
-   const searchGroupByColor = (searchColor, groupList) => {
-      return groupList
-         .map((group) => {
-            const { color } = group;
-
-            if (searchColor === color) {
-               return { ...group, color: searchColor };
-            } else {
-               return undefined;
-            }
-         })
-         .filter((groups) => groups !== undefined);
-   };
-
-   const setColor = (color) => {
-      if (color !== "white") {
-         const updatedGroup = searchGroupByColor(color, data);
-         setTodoArray(updatedGroup);
-      } else {
-         setTodoArray(data);
-      }
-   };
-
-   const addNewGroup = (newGroup) => {
-      const updatedGroupArray = [...todoGroupArray];
-      setTodoArray([...updatedGroupArray, newGroup]);
-   };
-
    return (
       <div className="dashboard">
          <div className="dashboardHeader">
             <div className="dashboardTodoSearch">
-               <TodoSearch onClear={onClear} onChange={onChange} value={searchValue} />
+               <TodoSearch value={searchValue} onChange={onChange} onClear={onClear} />
+
                <div className="dashboardColorPanel">
-                  <ColorPanel setColor={setColor} />
+                  <ColorPanel setColor={searchGroupByColor} />
                </div>
-               <TodoFilter show={showCompleted} />
+
+               <TodoFilter show={filterTodoByDone} />
+            </div>
+            <div>
+               <FilterByDate />
             </div>
          </div>
 
-         <AddGroup addNewGroup={addNewGroup} nextGroupId={todoGroupArray.length + 1} />
+         <AddGroup nextGroupId={groupList.length + 1} />
 
          <div className="dashboardTodoGroup">
-            {todoGroupArray.map((obj, index) => (
-               <TodoGroup
-                  key={index + obj.id}
-                  color={obj.color}
-                  title={obj.title}
-                  created={obj.created}
-                  todoList={obj.todoList}
-                  addTodo={addNewTodo}
-                  id={obj.id}
-                  handleCheckbox={handleCheckbox}
-               />
-            ))}
+            {groupList.map((groups, index) => {
+               if (
+                  (groups.sortByColor === null || groups.sortByColor === true) &&
+                  (groups.sortByCreated === null || groups.sortByCreated === true)
+               ) {
+                  return (
+                     <TodoGroup
+                        key={index + " " + groups.id}
+                        color={groups.color}
+                        title={groups.title}
+                        created={groups.created}
+                        id={groups.id}
+                        todoItems={groups.todoList}
+                        nextItemId={groups.todoList.length}
+                        todoCheckBox={todoCheckBox}
+                     />
```

Comment:
Please update setting props. to component. Use {...group } construction instead of passing every prop. separately.